ONSdigital/ras-rm-auth-service
# RAS-1127 Capture Respondent Account Verification Date-Time

# What and why?
Updating the database model to store and return the account verification date
# How to test?
- deploy this PR to your env
- run the acceptance tests
- get an IAC from rOps after running the acceptance tests
- create an account in frontstage using the IAC 
- check that the `account_verification_time` is `NULL` for the account you just created
- verify your account by getting the link from the party logs (updating the localhost port)
- check that the `account_verification_time` now has the timestamp added

Also 
- update the email (getting the verification link in the same way)
- check that the timestamp is updated
- verify that it does not change after a password reset
  - Including when the account becomes locked (after 10 failed logins) and the password is reset
- Check that the returned object when you go to the respondent details page contains the `account_verification_date` timestamp
  - I've done this by running a local version of rOps with a print on [this](https://github.com/ONSdigital/response-operations-ui/blob/1ec70c89e443fdfba620af328a4a13ce67459aa8/response_operations_ui/controllers/respondent_controllers.py#L40) line

Note: The acceptance tests do not cause auth to update the `account_verification_time` column because they don't actually use the link but go and hit a party endpoint to activate the user. This endpoint specifically does it by unlocking the account which we explicitly did not add as a path where the verification date would be updated because that path is _usually_ only used when a respondent resets their password, and are thus not verified unlike when they first activate their account or change their email
# Jira
https://jira.ons.gov.uk/browse/RAS-1127
